build2cmake/src/config/v3.rs

```diff
@@ -0,0 +1,412 @@
+use std::{
+    collections::{BTreeSet, HashMap},
+    fmt::Display,
+    path::PathBuf,
+    str::FromStr,
+};
+
+use eyre::Result;
+use itertools::Itertools;
+use serde::{Deserialize, Serialize};
+
+use crate::version::Version;
+
+use super::{common::Dependency, v2};
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct Build {
+    pub general: General,
+    pub torch: Option<Torch>,
+
+    #[serde(rename = "kernel", default)]
+    pub kernels: HashMap<String, Kernel>,
+}
+
+impl Build {
+    pub fn is_noarch(&self) -> bool {
+        self.kernels.is_empty()
+    }
+
+    pub fn supports_backend(&self, backend: &Backend) -> bool {
+        self.general.backends.contains(backend)
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct General {
+    pub name: String,
+
+    pub backends: Vec<Backend>,
+
+    pub cuda: Option<CudaGeneral>,
+
+    pub hub: Option<Hub>,
+
+    pub python_depends: Option<Vec<PythonDependency>>,
+}
+
+impl General {
+    /// Name of the kernel as a Python extension.
+    pub fn python_name(&self) -> String {
+        self.name.replace("-", "_")
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct CudaGeneral {
+    pub minver: Option<Version>,
+    pub maxver: Option<Version>,
```

Comment:
`cuda-minver` and `cuda-maxver` have moved from `general` to this new `general.cuda` section.